### ISSUE TITLE
Save: update transfer functions (ignore hash tables)

### ISSUE THREAD

**Issue by ellynliu** [uchicago-cs/chiventure#92]
Remove any placeholder structs and replace them with the updated game state structs. Use this and the new .proto file to fix/update all transfer functions in save.c that have been broken. Ignore anything with hash tables, as this will be an issue of its own.

**Comment by ellynliu** [comment]
So far updating the code to reflect changes in game state's structs has been pretty straightforward. I looked at both the .proto file and game state's header files to update previous functions and variable names and wrote new functions to save attribute_t, attribute_value_t, condition_t, and path_t.

**Comment by ellynliu** [comment]
Based off the team meeting discussion, although we may not to save condition_t in the future, since we are currently working on dumping out the entire contents of game state, these functions will still be relevant. Placeholder structs have been removed and functions that do not involve hash tables have been completed. A pull request to dev was completed so my code has been merged into dev. Now work on saving information contained in hash tables (in the functions that I just wrote) will be continued in the next issue.
